lib/vinyl-stream.js

```diff
@@ -35,6 +35,9 @@ module.exports = function createVinylStream(options) {
 				path: base + '/' + object.Key
 			});
 
+	  if(!file.contents)
+		  file.contents =  new Buffer("");
+
 		// Set the length on things like streams which are used
 		// by some frameworks.
 		if (!file.contents.length) {
```

Comment:
Maybe change this to `if (!file.isNull() && !file.contents.length) {` instead? I think that makes more sense since we don't want to be modifying existing vinyl objects (e.g. if we get a directly vinyl object we don't want to be giving it `contents` when it shouldn't have any).